src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/components/flyout/flyout.js

```diff
@@ -273,9 +273,15 @@ class FlyoutUI extends Component {
       confirmModalPromise
     );
 
-    const byId = groupBy(conflictedIndexPatterns, ({ obj }) =>
-      obj.searchSource.getOwnField('index')
-    );
+    const byId = {};
+    conflictedIndexPatterns
+      .map(({ doc, obj }) => {
+        return { doc, obj: obj._serialize() };
+      }).forEach(({ doc, obj }) =>
+        obj.references.forEach(ref => {
+          byId[ref.id] = byId[ref.id] != null ? byId[ref.id].concat({ doc, obj }) : [{ doc, obj }];
```

Comment:
why not add `conflictedIndexPatternsById` as a part of `resolveSavedObjects` api? this logic is not related to the component.

Reply:
it makes a lot of sense to move it, but when I do a lot of tests break :( because there's so much business logic in _serialize() it's not so easy to stub out.